### PR TITLE
[Polygon Edge] Add performance report from July

### PR DESCRIPTION
**Pull request by laviniat1996**

### CODE REVIEW

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+This test was done to measure the SC ERC20 token transfers, SC ERC721 token minting and EOA to EOA transactions functionality with heavy loads and speed of the transactions on the nodes with higher hardware resources.
+
+The goal was to check if everything is working as expected during heavy loads. That was also the reason we’ve introduced gas metrics in the loadbot output, which show us if the blocks are filled with transactions properly.
+
+All transactions were sent to the single node via GRPC API, and the receipts were received via JSON-RPC API. After all transactions were done, gas information was read from each block, using the eth_getBlockByNumber JSON-RPC method.
+
+Our aim was to strive to reach a maximum possible TPS on the hardware resources available.
+To acheive this, we've modified the block gas limit and block time parameters, to give us the best possible tps results and keep the system integrity and stability.
+
+:::info Block Gas Limit
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:26]: We can remove this info block

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+In the example stated bellow, ERC721 token minting worked much faster with a block gas limit set to 80 000 000 ( instead of 20 Mil. ), but with ERC20 token transfers with 80 Mil. block gas limit, the server crashed.
+:::
+
+:::info Production Environments
+When configuring a production environment, you need to be carefull if you're trying to acheive high performance of the chain.
+If block gas limit parameter is set to a high value, block time is set to 1s, and there is a high transaction load on a single node, that node will consume a lot ( if not all available ) RAM and can cause server crash.
+Use the loadbot to test everything thoroughly, monitor the system resource utilization and set your configuration parameters accordingly.
+:::
+
+:::info Out of Memory Errors
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:37]: We can remove this info block, as it was stated in the previous perf report

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+</details>
+
+<details>
+  <summary>Blockchain Configuration</summary>
+  <div>
+    <div>
+        <table>
+            <tr>
+                <td>Polygon Edge version</td>
+                <td>Commit <a href="https://github.com/0xPolygon/polygon-edge/commit/6bb316b1d99c229278076431e6586852ff4e36f9">6bb316b1d99c229278076431e6586852ff4e36f9</a> on develop branch</td>
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:268]: release version v0.4.1

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:441]: release v0.4.1

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+id: test-2022-07-04
+title: July 4th 2022
+description: Performance test from July 4th.
+keywords:
+  - docs
+  - polygon
+  - edge
+  - performance
+  - test
+  - EOA
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:11]: Maybe add ERC20 and ERC721 as keywords ?

**Review by ZeljkoBenovic — CHANGES_REQUESTED** [review]
Looks good 🥇 

There are just some minor changes required.

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 1 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed 3792e31b22137069761bc086f2582bddbc376fcc

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 2 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed 3792e31b22137069761bc086f2582bddbc376fcc

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+</details>
+
+<details>
+  <summary>Blockchain Configuration</summary>
+  <div>
+    <div>
+        <table>
+            <tr>
+                <td>Polygon Edge version</td>
+                <td>Commit <a href="https://github.com/0xPolygon/polygon-edge/commit/6bb316b1d99c229278076431e6586852ff4e36f9">6bb316b1d99c229278076431e6586852ff4e36f9</a> on develop branch</td>
```
[the remark this answers is not in the document]
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed f78879ba34e23cbc75edcfb243975402f89e8eeb

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 3 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed f78879ba34e23cbc75edcfb243975402f89e8eeb

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 4 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed f78879ba34e23cbc75edcfb243975402f89e8eeb

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 5 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md:11]: fixed d48807185f9052f53eeb4b6981506abf9a2d4d62

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 7]
```diff
@@ -124,7 +114,7 @@ Mar 23 00:19:06 ip-10-151-2-196 kernel: oom-kill:constraint=CONSTRAINT_NONE,node
         <table>
             <tr>
                 <td>Polygon Edge version</td>
-                <td>Commit <a href="https://github.com/0xPolygon/polygon-edge/commit/6bb316b1d99c229278076431e6586852ff4e36f9">6bb316b1d99c229278076431e6586852ff4e36f9</a> on develop branch</td>
+                <td><a href="https://github.com/0xPolygon/polygon-edge/releases/tag/v0.4.1">v0.4.1</a> on develop branch</td>
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:117]: This should probably be: `<td>Release <a href="https://github.com/0xPolygon/polygon-edge/releases/tag/v0.4.1">v0.4.1</a> </td>`

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 8]
```diff
@@ -265,7 +255,7 @@ Mar 23 00:19:06 ip-10-151-2-196 kernel: oom-kill:constraint=CONSTRAINT_NONE,node
...
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:258]: Same as other comment

File: docs/edge/performance-reports/test-history/test-2022-07-04.md [hunk 9]
```diff
@@ -438,7 +428,7 @@ Mar 23 00:19:06 ip-10-151-2-196 kernel: oom-kill:constraint=CONSTRAINT_NONE,node
...
```
Comment (ZeljkoBenovic) [inline docs/edge/performance-reports/test-history/test-2022-07-04.md:431]: Same as other comment

**Review by ZeljkoBenovic — CHANGES_REQUESTED** [review]

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 7 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed e4eeae9b3235cc805c0c0de0b2fe0686b711dd7a

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 8 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed e4eeae9b3235cc805c0c0de0b2fe0686b711dd7a

File: docs/edge/performance-reports/test-history/test-2022-07-04.md (hunk 9 above)
Comment (laviniat1996) [reply docs/edge/performance-reports/test-history/test-2022-07-04.md]: fixed e4eeae9b3235cc805c0c0de0b2fe0686b711dd7a

**Review by DannyS03 — APPROVED** [review]
thanks folks.

we'll add the team as contributors to the repo so contributions can be direct

**Pull request merged**